viewing_party/party.py

```diff
@@ -1,23 +1,196 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+# if title or genre or rating is false, return None
+    if not title or not genre or not rating:
+        return None 
+
+# if inputs are true, return dictionary{title: ; genre: ; rating: }
+    # movie = {}
+    # movie["title"] = title
+    # movie["genre"] = genre
+    # movie["rating"] = rating
+
+    movie = {
+        "title" : title,
+        "genre": genre,
+        "rating": rating
+    }
+    return movie
+
+def add_to_watched(user_data, movie):
+# user_data = {watched: [{}]}, if no movies watched, then empty list.
+# movie =  {
+#         "title": "Title A",
+#         "genre": "Horror",
+#         "rating": 3.5
+#       }
+# add movie to watched in user_data. Return user_data.
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+# user_data = {watchlist: [{}]}, if no movies in watchlist, then empty list.
+# movie =  {
+#         "title": "Title A",
+#         "genre": "Horror",
+#         "rating": 3.5
+#       }
+# add movie to watchlist in user_data. Return user_data.
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+# user_data = {watchlist: ; watched: }
+# title = "movie user has watched"
+# if title is in watchlist, remove from watchlist, add to watched, return user_data.
+# if title not in watchlist, return user_data.
+
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+            
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    total_rating = 0.0
+
+    if not user_data["watched"]:
+        return total_rating
+    
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"]
+    
+    average_rating = total_rating / len(user_data["watched"])
+    
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    
+    if user_data["watched"] == []:
+        return None
+    
+    genre_dict = {}
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_dict:
+            genre_dict[genre] += 1
+        else:
+            genre_dict[genre] = 1
+        
+    highest_counter = 0
+
+    for genre in genre_dict:
+        if genre_dict[genre] > highest_counter:
+            highest_counter = genre_dict[genre]
+            most_frequent_genre = genre

```

Comment:
Don't forget that we have our `.items()` method for Python dictionaries! This could allow us to streamline this genre counter:

```suggestion
    for genre, count in genre_dict.items():
        if count > highest_counter:
            highest_counter = count
            most_frequent_genre = genre

```